admin/server/auth/handlers.go

```diff
@@ -324,6 +427,97 @@ func (a *Authenticator) authWithToken(w http.ResponseWriter, r *http.Request) {
 	http.Redirect(w, r, redirect, http.StatusTemporaryRedirect)
 }
 
+func (a *Authenticator) authAssumeOpen(w http.ResponseWriter, r *http.Request) {
+	sess := a.cookies.Get(r, cookieName)
+	authToken, ok := sess.Values[cookieFieldAccessToken].(string)
+	if !ok || authToken == "" {
+		http.Error(w, "not authenticated: please log in first", http.StatusUnauthorized)
+		return
+	}
+	ctx := r.Context()
+	// Validate token against database
+	validated, err := a.admin.ValidateAuthToken(ctx, authToken)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	claims := newAuthTokenClaims(validated, a.admin)
+	if claims.OwnerType() != OwnerTypeUser {
+		http.Error(w, "not authenticated as a user", http.StatusUnauthorized)
+		return
+	}
```

Comment:
If you wrap this handler with the auth middleware, I think you can simplify this similar to here: https://github.com/rilldata/rill/blob/7caad7ca93be914fc318727805bba718c6f1ff33/admin/server/auth/device_code.go#L129